qiskit/algorithms/fidelities/fidelity_result.py

```diff
@@ -0,0 +1,32 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""
+Fidelity result class
+"""
+
+from __future__ import annotations
+
+from typing import Any
+from dataclasses import dataclass
+
+
+@dataclass(frozen=True)
+class FidelityResult:
+    """Result of Fidelity computation.
+
+    Args:
+        values (list[float]): List of fidelity values for each pair of input circuits.
```

Comment:
Would `fidelities` here as a list[float] be a more descriptive field name than plain `values`

Reply:
I think it would be good if the result object doesn't include any logic if possible, so truncating the fidelities might be best left to the algorithm 🤔 

Maybe `fidelities` is better yes, as the result in the sampler is called `quasi_dists` and not `values`. The gradient result should also contain a `gradients` field then 😄

Reply:
>  truncating the fidelities might be best left to the algorithm

I meant more that the I should be able to run a fidelity and use the result I get back consistently. What are you thinking of in terms of truncation here - as far as I can see an algorithm can do what it wants with these values. Are you saying maybe you can get 1.000001 or something or -0.00003 I think we need to say something to make an algorithm aware of what to expect, and what any derived fidelity class should return back.

Reply:
It would be nice to have `truncate_fidelity() -> list[float]` methods that ensure that the value is between 0 and 1.

Reply:
Alright, I will change the name to `fidelities` (I have no strong opinions in that matter), and I will look into `truncate_fidelity()`.

Reply:
Understood. The `StateFidelityResult` class has now a `fidelities` and `raw_fidelities` field, where the fidelities are always [0,1] and the raw ones might not be.